mkt/webapps/indexers.py

```diff
@@ -423,11 +424,17 @@ def run_indexing(cls, ids, ES=None, index=None, **kw):
 
         qs = Webapp.with_deleted.no_cache().filter(id__in=ids)
         ES = ES or cls.get_es()
-
         docs = []
-        for obj in list(qs):
+        for i, obj in enumerate(qs):
             try:
                 docs.append(cls.extract_document(obj.id, obj=obj))
+            except SoftTimeLimitExceeded:
+                from lib.es.management.commands import reindex
+                # obviously this chunk was too big, maybe try two smaller ones.
+                mid = len(ids) / 2
+                reindex.run_indexing.apply_async(args=[index, cls, ids[:mid]])
```

Comment:
My understanding was the soft time limit was a chance to finish/clean up, but the hard time limit force kills the task.